### ISSUE TITLE
Can't umount dislocker file

### ISSUE THREAD

**Issue by bprimrose** [Aorimn/dislocker#136]
Windows is the official company OS, I'm running Ubuntu 17.10 in dual-boot and  I need to back my files up to the Windows filesystem to avoid the ire of Corp IT.

I mounted via sudo and say:/
```
    $ sudo dislocker -r -V /dev/nvme0n1p2 -p586036-270226-236071-434709-493537-192544-225907-208329 -- /media/bitlocker
    $ sudo mount -r -o loop,uid=1000,gid=1000 /media/bitlocker/dislocker-file /media/doze
```

Attempt to copy files, find out I mounted RO via a screenful of errors:
`    $ rsync --delete -ta Documents/ /media/doze/Users/bprimrose/Documents/`

No biggy, dismount and remount RW
```
    $ sudo umount /media/doze
    $ sudo umount /media/bitlocker/dislocker-file 
    umount: /media/bitlocker/dislocker-file: not mounted.
```

Oh, it's not mounted.  Great, let's remount:
    ```
$ sudo ls -l /media/bitlocker/dislocker-file
    -r--r--r-- 1 root root 145600019968 Dec 31  1969 /media/bitlocker/dislocker-file
    $ sudo dislocker -V /dev/nvme0n1p2 -p586036-270226-236071-434709-493537-192544-225907-208329 -- /media/bitlocker
    fuse: mountpoint is not empty
    fuse: if you are sure this is safe, use the 'nonempty' mount option
```

Can anyone here explain *why* I can't unmount the dislocker file, and *how* I can?  Is it safe to shut down my system at this point?

**Comment by superbonaci** [comment]
If you mounted it read only you can force close.

**Comment by thinrope** [comment]
Please do NOT post sensitive data (recovery passwords), your "Corp IT" will spank you :-D

**Comment by tlbprim** [comment]
That was really out of character for me.  Any way to edit a former comment to remove the info?

**Comment by thinrope** [comment]
Yes, you should be able to edit your comments (the pen icon top right of the comment box), but once leaked on the Net, data is cashed here and there and in at least 10 other places... The usual way is to change a leaked password, and you may have other policies (e.g. notify "Corp IT")...

**Comment by Aorimn** [comment]
For the initial problem, I've no solution indeed, but doing a `sync` then force unmount will help mitigate issues with data corruption.
On the other hand, using sometimes `fusermount -u /media/bitlocker/dislocker-file` will work while `umount` won't, not sure why: I've never been able to reproduce the issue consistently.

**Comment by paulstelian97** [comment]
Maybe you need to do `losetup -d` on where you have bound dislocker-file? If you did it manually. Mounting with -o loop should automatically clean the loop device setup but please verify it.

**Comment by bprimrose** [comment]
The laptops wiped now (I'm on to a new job).  I'll pick this back up if I
dual boot at the new one.

**Comment by superbonaci** [comment]
stop quoting private messages and emails

**Comment by yechs** [comment]
I have experienced the exact same error.

After checking with `mount` command, I found the following:
```console
# mount
[redacted]
dislocker on /media/bitlocker type fuse.dislocker (rw,nosuid,nodev,relatime,user_id=0,group_id=0)
```

So instead of un-mounting the `dislocker-file` as illustrated in the man page, what worked for me was to un-mount the parent folder: `umount /media/bitlocker`

**Comment by superbonaci** [comment]
@yechs exactly you umount the directory containing the `dislocker-file` not the file itself.
